trinity/__init__.py

```diff
@@ -1,14 +1,46 @@
-import pkg_resources
+import os
+import re
+import subprocess
 import sys
 
+import pkg_resources
+
+
+def read(*parts: str) -> str:
+    try:
+        with open(os.path.join(*parts), 'r') as f:
+            return f.read()
+    except IOError:
+        return ''
+
+
+def find_version(*file_paths: str) -> str:
+    version_file = read(*file_paths)
+    version_match = re.search(r"version=['\"]([^'\"]*)['\"]", version_file)
+    if version_match:
+        return version_match.group(1)
+    return ''
+
+
+def get_git_commit_id(path: str) -> str:
+    try:
+        return subprocess.check_output(['git', '-C', path, 'rev-parse', 'HEAD'],
+                                       stderr=subprocess.DEVNULL, encoding='UTF-8')
+    except subprocess.CalledProcessError:
+        return ''
```

Comment:
I think I'd rather the exception be handled at the call site than within this function.  Otherwise there is *some* ambiguity as to whether the subprocess check actually *somehow* returned an empty string or whether the call threw an exception.  Minor semantics, but I think it's a slightly better approach.